controllers/namespace/namespace_controller.go

```diff
@@ -0,0 +1,194 @@
+package namespace
+
+import (
+	"context"
+	"fmt"
+	"strings"
+	"sync"
+
+	"github.com/go-logr/logr"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/klog/v2"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+// once is a pointer to the sync.Once type
+var once sync.Once
+
+// funcError contains the error that occurred inside the once.Do() function call
+var funcErr error
+
+// controllerInit is a bool that denotes if the controller is reconciling for the first time
+var controllerInit bool
```

Comment:
I think we dont need controllerInit at all, It is making things complicated unnecessarily.

Reply:
It is used to avoid refetching of the namespace name, annotations if the operator is reconciling for the first time.
Or, we are fine with just one extra fetch for the first reconcile? Or, we can handle it in some other way?

Reply:
I think it is okay because that will happen just for once.